src/ContractInteract/OSTPrime.js

```diff
@@ -0,0 +1,248 @@
+      return Promise.reject(err);
+    }
+    if (typeof amount !== 'string') {
+      const err = new TypeError('Invalid approval amount.');
+      return Promise.reject(err);
+    }
+    const tx = this.contract.methods.approve(spenderAddress, amount);
+    return Promise.resolve(tx);
+  }
+
+  /**
+   * Returns the allowance amount for the given account address
+   *
+   * @param {string} ownerAddress Owner account address.
+   * @param {string} spenderAddress Spender account address.
+   *
+   * @returns {Promise} Promise object.
+   */
+  allowance(ownerAddress, spenderAddress) {
+    if (!Web3.utils.isAddress(ownerAddress)) {
+      const err = new TypeError('Owner address is invalid or missing');
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(spenderAddress)) {
+      const err = new TypeError('Spender address is invalid or missing');
+      return Promise.reject(err);
+    }
+    return this.contract.methods
+      .allowance(ownerAddress, spenderAddress)
+      .call()
+      .then((allowance) => {
+        return allowance;
+      });
+  }
+
+  /**
+   * Check if the account has approved gateway contract.
+   *
+   * @param {string} ownerAddress Owner account address.
+   * @param {string} spenderAddress Spender account address.
+   * @param {string} amount Approval amount.
+   *
+   * @returns {bool} `true` if approved.
+   */
+  isAmountApproved(ownerAddress, spenderAddress, amount) {
+    if (!Web3.utils.isAddress(ownerAddress)) {
+      const err = new TypeError('Invalid owner address.');
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(spenderAddress)) {
+      const err = new TypeError('Invalid spender address.');
+      return Promise.reject(err);
+    }
+    return this.allowance(ownerAddress, spenderAddress).then(
+      (approvedAllowance) => {
+        return new BN(amount).lte(new BN(approvedAllowance));
+      },
+    );
+  }
+
+  /**
+   * Returns the balance of an account.
+   *
+   * @param {string} accountAddress Account address
+   * @return {Promise} Promise object.
+   */
+  balanceOf(accountAddress) {
+    if (!Web3.utils.isAddress(accountAddress)) {
+      const err = new TypeError('Invalid address.');
+      return Promise.reject(err);
+    }
+    return this.contract.methods.balanceOf(accountAddress).call();
+  }
+
+  /**
+   * Unwrap amount.
+   *
+   * @param {string} amount Amount to unwrap.
+   * @param {Object} txOptions Transaction options.
+   *
+   * @return {Promise} Promise object.
+   */
+  unwrap(amount, txOptions) {
+    if (!txOptions) {
+      const err = new TypeError('Invalid transaction options.');
+      return Promise.reject(err);
+    }
+    return this._unwrapRawTx(amount).then((tx) =>
+      Utils.sendTransaction(tx, txOptions),
+    );
+  }
+
+  /**
+   * Unwrap amount raw tansaction.
+   *
+   * @param {string} amount Amount to unwrap.
+   *
+   * @return {Promise} Promise object.
+   */
+  _unwrapRawTx(amount) {
```

Comment:
Private methods should be documented as such via JSDoc: http://usejsdoc.org/tags-private.html

Reply:
If they are not supposed to be private functions, then the new functions we introduce shouldn't have an underscore prefix, and we should migrate the other functions accordingly.

@sarvesh-ost @schemar

Reply:
I too agree that underscore was an indication for private functions.
I can change all the functions in my next ticket.

Reply:
👍

Reply:
Yes. Can you find out _why_ they have an underscore @deepesh-kn ?